drivers/i2c/i2c_nrfx_twi.c

```diff
@@ -103,7 +109,24 @@ static int i2c_nrfx_twi_transfer(struct device *dev, struct i2c_msg *msgs,
 			}
 		}
 
-		k_sem_take(&(get_dev_data(dev)->completion_sync), K_FOREVER);
+		res = k_sem_take(&(get_dev_data(dev)->completion_sync),
```

Comment:
`res` is of type `nrfx_err_t` (which is by default an enum but potentially can be defined to something else) so it's not suitable for the return value of `k_sem_take`.

Reply:
Changed to use `ret` instead.